client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -80,17 +89,37 @@ interface JsonFormOptions {
 }
 
 export const useJsonForms = (
-  docName: string | undefined,
+  docName: string,
   options: JsonFormOptions = {}
 ) => {
-  const [data, setData] = useState<JsonData>(patient); // Replace with DB query hook
-  const [loading] = useState(false); // Replace with DB query hook
+  const { data: databaseResponse, isLoading } =
+    useDocument.document.get(docName);
+  const [data, setData] = useState<JsonData | undefined>();
+  const [jsonSchema, setJsonSchema] = useState<JsonSchema | undefined>();
+  const [uiSchema, setUiSchema] = useState<UISchemaElement | undefined>();
   const [saving, setSaving] = useState(false);
   const [error, setError] = useState<string | false>(false);
+  const [isDirty, setIsDirty] = useState<boolean>();
   const t = useTranslation('common');
   const { success, error: errorNotification } = useNotification();
   const navigate = useNavigate();
-  const { isDirty, setIsDirty } = useDirtyCheck();
+
+  useEffect(() => {
+    if (!databaseResponse) return;
+
+    const {
+      data,
+      // @ts-ignore
+      documentRegistry: { jsonSchema, uiSchema },
+    } = databaseResponse;
```

Comment:
It's complaining because `documentRegistry` can be null | undefined, in which case the destructuring will fail.

```suggestion
    if (!databaseResponse) return;

    const { data, documentRegistry } = databaseResponse;
    const { jsonSchema, uiSchema } = documentRegistry ?? {};

```

Reply:
I started a branch from this PR and need to make some changes regarding documentRegistry. I fixed it along the way as well, so better wait for my PR...